course-instructions.md

```diff
@@ -0,0 +1,760 @@
+
+---
+
+Another example of when this object (rather than target) pattern comes in handy is when we want to _force_ a target to rebuild. Note that in the example below, we are writing the `command` for this target by putting two lines of code between `{}` rather than calling a separate custom function. You can do this for any target, but it is especially useful in this application when we just have two lines of code to execute.
+
+```r
+library(targets)
+
+list(
+  tar_target(
+    work_files,
+    {
+      dummy <- '2021-04-19'
+      item_file_download(sb_id = "4f4e4acae4b07f02db67d22b", 
+                         dest_dir = "1_fetch/tmp",
+                         overwrite_file = TRUE)
+    },
+    format = "file",
+    packages = "sbtools"
+  )
+)
+```
+By adding this `dummy` object to our `command` argument for the `work_files` target, we can modify the dummy contents any time we want to force the update of `work_files`. Updating the `dummy` argument to today's date allows us to simultaneously force the update and record when we last downloaded the data from ScienceBase. You may see the use of these `dummy` arguments in spots where there is no other trigger that would cause a rebuild, such as pulling data from a remote webservice or website when `targets` has no way of knowing that new data are available on the same service URL.
+
+---
+
+</details>
+
+<hr>
+
+<details><summary><h2>Strategies for defining targets in data pipelines</h2></summary>
+
+### How to make decisions on how many targets to use and how targets are defined
+
+We've covered a lot of content about the rules of writing good pipelines, but pipelines are also very flexible! Pipelines can have as many or as few targets as you would like, and targets can be as big or as small as you would like. The key theme for all pipelines is that they are reproducible codebases to document your data analysis process for both humans and machines. In this next section, we will learn about how to make decisions related to the number and types of targets you add to a pipeline.
+
+### Background 
+Isn't it satisfying to work through a fairly lengthy data workflow and then return to the project and it _just works_? For the past few years, we have been capturing the steps that go into creating results, figures, or tables appearing in data visualizations or research papers. There are recipes for reproducibility used in complex, collaborative modeling projects, such as in [this reservoir temperature modeling pipeline](https://code.usgs.gov/wma/wp/res-temperature-process-models) and in [this pipeline to manage downloads of forecasted meteorological driver data](https://code.usgs.gov/wma/wp/forecasted-met-drivers). _Note that you need to be able to access internal USGS websites to see these examples and these were developed early on in the Data Science adoption of `targets` so may not showcase all of our adopted best practices_.
```

Comment:
This should be updated to reflect current work as well.

Reply:
#29